src/Foundation/NSDate.cs

```diff
@@ -42,10 +42,19 @@ public partial class NSDate {
 		public static explicit operator DateTime (NSDate d)
 		{
 			double secs = d.SecondsSinceReferenceDate;
-			if ((secs < -63113904000) || (secs > 252423993599))
-				throw new ArgumentOutOfRangeException ("Value is outside the range of NSDate");
 
-			return new DateTime ((long)(secs * TimeSpan.TicksPerSecond + NSDATE_TICKS), DateTimeKind.Utc);
+			if ((secs < -63114076800) || (secs > 252423993599))
+				throw new ArgumentOutOfRangeException ($"Value is outside the range of NSDate {secs}");
+
+			var calendar = new NSCalendar (NSCalendarType.Gregorian);
+			calendar.TimeZone = NSTimeZone.FromName ("UTC");
+			NSDateComponents components = calendar.Components (NSCalendarUnit.Year | NSCalendarUnit.Month | NSCalendarUnit.Day | NSCalendarUnit.Hour |
```

Comment:
Components are also IDisposable.

Reply:
mixed emotions here
This method can be called multiple times, e.g. in a loop, so it can be costly do recreate the same `NSCalendar` object every time.
I think it would be better to cache the `calendar` to a static member of the class.

Reply:
If you're going to make `components` a singleton, be aware that lazy singletons have thread-safety issues. Jon Skeet wrote up a fairly exhaustive description of that [here](https://csharpindepth.com/articles/singleton).

Reply:
@stephen-hawley I'm +1 on the use of a static constructor on the NSDate class. It will just be executed:

* An instance of the class is created.
* Any of the static members of the class are referenced.

In our case, the most probable cause is that the native apps create the instance of the class, later to be casted by the user. The purely managed code that just uses DateTime will not be affected and if the NSDate is never instantiated of the casting are never used, the static constructor won't be called. 

In this case we are not working with a Singleton, so doing Lazy<T> for the calendar and the component is going to be an overkill AND a lot more complicated since NSComponents depends on the NSCalendar, so you have one lazy init with a callback that depends on an lazy init with a second callback. 

I wonder if we could look at the AOT results, but I think the static constructor is probably something easier to AOT that the two Lazy templates with two delegates.

Reply:
Is the size increase for AOT really that big for NSDateComponents? I would not even do a singleton or lazy, it looks as simple as creating a `static readonly` field which does not have to check if a static constructor has been called on every field access.

Reply:
With ThreadLocal, we get 'lazy'. The NSCalendar can be static.